ipatests/test_ipaplatform/test_constants.py

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+from ipaplatform.base.constants import User, Group
+
+
+def test_user():
+    user = User("root")
+    assert user == "root"
+    assert str(user) == "root"
+    assert repr(user) == '<User "root">'
+    assert user.uid == 0
+    assert user.pgid == 0
+    assert user.entity.pw_uid == 0
+
+
+def test_group():
+    group = Group("root")
+    assert group == "root"
+    assert str(group) == "root"
+    assert repr(group) == '<Group "root">'
+    assert group.gid == 0
+    assert group.entity.gr_gid == 0
```

Comment:
This might be distro-specific but in Fedora we can predict the uid of other users, like httpd and dirsrv. Would it be better to check less trivial users as well? Maybe as a separate test with a skip?

Reply:
And a test for a non-existent user and group is needed. It doesn't currently work very nicely. It may be preferable to raise an exception when a user can't be located to make debugging easier.

```
>>> from ipaplatform.base.constants import BaseConstantsNamespace, User, Group
>>> foo = User("notfound")
>>> foo.uid
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rcrit/redhat/freeipa/ipaplatform/base/constants.py", line 46, in uid
    return self.entity.pw_uid
  File "/home/rcrit/redhat/freeipa/ipaplatform/base/constants.py", line 39, in entity
    self._entity = entity = pwd.getpwnam(self)
KeyError: 'getpwnam(): name not found: <User "notfound">'
```